crates/hir-def/src/visibility.rs

```diff
@@ -94,15 +94,15 @@ impl RawVisibility {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub enum Visibility {
     /// Visibility is restricted to a certain module.
-    Module(ModuleId),
+    Module(ModuleId, ModuleVisibilityMeta),
```

Comment:
Let's swap that for
```suggestion
    Module(ModuleId, VisibilityExplicity),
```
and
```rs
enum VisibilityExplicity {
    Explicit,
    Implicit,
}
```
Whether the import is `pub(crate)` or not can be inferred from the flag being `Explicit` and the `ModuleId` being a root module of the same crate. That is we can be more general here.

Reply:
Alternatively
```suggestion
    Module { module: ModuleId, is_explicit: bool },
```
choose whatever you prefer